instant-distance/src/lib.rs

```diff
@@ -301,78 +283,83 @@ where
             done: AtomicUsize::new(0),
         };
 
-        for (layer, range) in ranges {
+        for layer in state.meta.descending() {
             #[cfg(feature = "indicatif")]
             if let Some(bar) = &state.progress {
                 bar.set_message(format!("Building index (layer {})", layer.0));
             }
 
-            let inserter = |pid| state.insert(pid, layer, &layers);
-
-            let end = range.end;
-            if layer == top {
-                range.into_iter().for_each(|i| inserter(PointId(i as u32)))
-            } else {
-                range
-                    .into_par_iter()
-                    .for_each(|i| inserter(PointId(i as u32)));
-            }
+            let Range { start, end } = state.meta.points(layer);
+
+            // FIXME:
+            // There is some sort of race here where neighbors is missing nodes.
```

Comment:
We won't be able to merge this. Can you debug the issue?

Reply:
I plan to spend some more time on this.

Reply:
I resolved this (in [74d697c](https://github.com/instant-labs/instant-distance/pull/47/commits/74d697ce078f9c2f577e14a0d5f70ff80046fc6d))